src/Options/Auto.hpp

```diff
@@ -0,0 +1,64 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <optional>
+#include <ostream>
+#include <string>
+
+#include "Options/Options.hpp"
+
+namespace Options {
+/// \ingroup OptionParsingGroup
+/// A class indicating that a parsed value can be automatically
+/// computed instead of specified.
+///
+/// When an `Auto<T>` is parsed from an input file, the value may be
+/// specified either as "Auto" or as a value of type `T`.  When this
+/// class is passed to the constructor of the class taking it as an
+/// option, it can be implicitly converted to a `std::optional<T>`.
+///
+/// \snippet Test_Auto.cpp example_class
+/// \snippet Test_Auto.cpp example_create
+template <typename T>
+class Auto {
+ public:
+  Auto() = default;
```

Comment:
noexcept?

Reply:
The meaning of writing noexcept there varies between versions of the standard, and clang implements it incorrectly for C++17, so I'm avoiding it these days.  Leaving it out always does the right thing.  (Same with constexpr.)